src/core/src/package_managers/TdnfPackageManager.py

```diff
@@ -89,6 +94,31 @@ def refresh_repo(self):
         self.invoke_package_manager(self.cmd_clean_cache)
         self.invoke_package_manager(self.cmd_repo_refresh)
 
+    # region Strict SDP using SnapshotTime
+    def __get_posix_time(self, datetime_to_convert, env_layer):
```

Comment:
Not convinced this function needs to exist at all. 

I'd just write this in the constructor** where the function is currently called:

self.snapshot_posix_time = None if self.execution_config.maxpatchpublishdate == str() else self.env_layer.datetime_string_to_posix_time ...

If the conversion fails, it's okay to fail.

** = the other comment of that conversion only getting done in the install patches flow to avoid failures in the CP, AP.